explorer/src/components/account/BlockHistoryCard.tsx

```diff
@@ -0,0 +1,142 @@
+import React from "react";
+import { TableCardBody } from "components/common/TableCardBody";
+import { useBlock, useFetchBlock, FetchStatus } from "providers/block";
+import { Signature } from "components/common/Signature";
+import { ErrorCard } from "components/common/ErrorCard";
+import { LoadingCard } from "components/common/LoadingCard";
+import { Block } from "components/common/Block";
+
+export function BlockHistoryCard({ block }: { block: string }) {
+  const blockData = useBlock(block);
+  const fetchBlock = useFetchBlock();
+  const refresh = () => fetchBlock(block);
+
+  React.useEffect(() => {
+    if (!blockData) refresh();
+  }, [blockData, block]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  if (!blockData) {
+    return null;
+  }
+
+  if (blockData?.data === undefined) {
+    if (blockData.status === FetchStatus.Fetching) {
+      return <LoadingCard message="Loading block" />;
+    }
+
+    return <ErrorCard retry={refresh} text="Failed to fetch block" />;
+  }
+
+  if (blockData.status === FetchStatus.FetchFailed) {
+    return <ErrorCard text={`Block "${block}" is not valid`} />;
+  }
```

Comment:
Right now, the web3 doesn't provide a good way to differentiate between a bad request and a missing block. Using the same error message is fine: "Failed to fetch block"